src/ray/core_worker/context.cc

```diff
@@ -14,20 +14,28 @@ struct WorkerThreadContext {
 
   const TaskID &GetCurrentTaskID() const { return current_task_id; }
 
-  void SetCurrentTask(const TaskID &task_id) {
+  std::shared_ptr<const raylet::TaskSpecification> GetCurrentTask() const {
+    return current_task;
+  }
+
+  void SetCurrentTaskId(const TaskID &task_id) {
     current_task_id = task_id;
     task_index = 0;
     put_index = 0;
   }
 
   void SetCurrentTask(const raylet::TaskSpecification &spec) {
-    SetCurrentTask(spec.TaskId());
+    SetCurrentTaskId(spec.TaskId());
+    current_task = std::make_shared<const raylet::TaskSpecification>(spec);
```

Comment:
The reference will be invalid finally. Java uses JNI to update the current task, and the `raylet::TaskSpecification` instance is a short-lived one because Java passes serialized protobuf data and then deserialized in JNI. Maybe we can change it to an `std::shared_ptr`?

Reply:
oh, I see. In this PR, `SetCurrentTask` is called from Java. But after we migrate the whole Java worker on top of core worker, `SetCurrentTask` should be called by `CoreWorkerTaskExecutionInterface` when it starts executing a task, and `SetCurrentTask(nullptr)` should be called when that task finishes. So the task spec object should be valid for this whole process. Then, this can be a reference wrapper. Is this correct?

Reply:
You are right. But you can't pass a `nullptr` here unless we change the parameter to a pointer.